ras_party/controllers/respondent_controller.py

```diff
@@ -125,17 +125,64 @@ def update_respondent_mark_for_deletion(email: str, session):
 @with_db_session
 def delete_respondents_marked_for_deletion(session):
     """
-    Deletes all the existing respondents and there associated data which are marked for deletion
+    Deletes all the existing respondents and their associated data which are marked for deletion
+
+    NOTE: We don't delete pending_surveys records as these are subject to their own scheduled deletion
+    An IntegrityError exception will be logged if the respondent record cannot be deleted due
+    to existing pending_surveys records.
 
     :param session A db session
     """
     respondents = session.query(Respondent).filter(Respondent.mark_for_deletion == True)  # noqa
+    logger.info("Preparing to delete respondent records", number_to_delete=respondents.count())
```

Comment:
Very robust logging going on here. With party the way it is I think we should look to make this a little more streamlined, the attempting bit in my opinion can be removed, the is/isn't deleted covers this. Personally I would focus just on failures and worry less about the count

Reply:
Yeah, agreed. But this is very much to expose the extent of the issue, and any other scenarios that we might be suffering from. Currently we have zero sight of the extent of the problem. This can be changed to debug after a period of time once we're happy.